internal/service/accesslistapikey/resource_access_list_api_key.go

```diff
@@ -95,20 +95,19 @@ func resourceMongoDBAtlasAccessListAPIKeyCreate(ctx context.Context, d *schema.R
 		entry = IPAddress
 	}
 
-	createReq := matlas.AccessListAPIKeysReq{}
-	createReq.CidrBlock = CIDRBlock
-	createReq.IPAddress = IPAddress
-
-	createRequest := []*matlas.AccessListAPIKeysReq{}
-	createRequest = append(createRequest, &createReq)
+	accessList := &[]admin.UserAccessList{
+		{
+			CidrBlock: conversion.StringPtr(CIDRBlock),
+			IpAddress: conversion.StringPtr(IPAddress),
+		},
+	}
 
-	_, resp, err := conn.AccessListAPIKeys.Create(ctx, orgID, apiKeyID, createRequest)
+	_, resp, err := connV2.ProgrammaticAPIKeysApi.CreateApiKeyAccessList(context.Background(), orgID, apiKeyID, accessList).Execute()
```

Comment:
Fortunately this is not CFN :)
```suggestion
	_, resp, err := connV2.ProgrammaticAPIKeysApi.CreateApiKeyAccessList(ctx, orgID, apiKeyID, accessList).Execute()
```

Reply:
thx, i missed that one 😅